R/summarise_country.R

```diff
@@ -0,0 +1,163 @@
+summarise_sheet_indicator_list <- function(billion) {
+  billion <- rlang::arg_match(billion)
+  indicator_order_b <- indicator_order %>%
+    dplyr::filter(ind %in% billion_ind_codes(billion)) %>%
+    dplyr::arrange(order) %>%
+    dplyr::select(sdg, short_name, medium_name, unit_raw, )
+
+  return(indicator_order_b)
+}
+
+#' Summarise Excel sheet 'data' for HPOP country summary
+#'
+#' `summarise_HPOP_cntry_data` summarises the 'data' sgeet for the HPOP country summary.
+#'
+#' @param iso3 ISO3 codes of country to export.
+#' @param ... Additional arguments passed to `calculate_hpop_contributions` and
+#' `calculate_hpop_billion`. Use if you need to provide additional parameters to
+#' those functions.
+#' @inherit calculate_hpop_contributions params
+#'
+#' @return list of data frames to be used in the 'data' sheet of HPOP country summary
+#' export
+#'
+#' @export
+summarise_HPOP_cntry_data <-
+  function(df,
+           iso3,
+           start_year = 2018,
+           end_year = 2019:2023,
+           ...) {
+    assert_mart_columns(df)
+    assert_years(start_year, end_year)
+
+    # Filter df for country, arrange indicators by order.
+    df_iso3 <- df %>%
+      dplyr::filter(iso3 == !!iso3) %>%
```

Comment:
I see the problem here. We should continue to refer to `iso3` simply as the column name to be consistent across all other functions. Please see my comment just below about strings as column names, but this should be explicitly passed to the function and then refered to in filter using `.data[[]]`. For sake of example, say we change the selection of countries to produce for to be `countries`, then we would have:

`dplyr::filter(.data[[iso3]] == !!countries)`

Reply:
Yes, I noticed that yesterday while cleaning a bit, and I have changed `iso3` in that context to simply `iso`.